src/test/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoublyLinkedListTest.java

```diff
@@ -0,0 +1,242 @@
+package org.fogbowcloud.manager.core.models.linkedlist;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import org.fogbowcloud.manager.core.models.orders.ComputeOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SynchronizedDoublyLinkedListTest {
+
+	private SynchronizedDoublyLinkedList list;
+	
+	@Before
+	public void initialize() {
+		this.list = new SynchronizedDoublyLinkedList();
+	}
+	
+	@Test
+	public void testAddFirst() {
+		assertNull(this.list.getHead());
+		assertNull(this.list.getTail());
+		
+		Order order = createOrder("0");
+		this.list.addItem(order);
+
+		assertEquals(order, this.list.getHead().getOrder());
+		assertEquals(order, this.list.getCurrent().getOrder());
+		assertEquals(order, this.list.getTail().getOrder());
+	}
+	
+	@Test
+	public void testAddOrder() {
+		assertNull(this.list.getNext());
+		
+		Order orderOne = createOrder("one");
+		Order orderTwo = createOrder("two");
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		// Next node of the second order is null.
+		assertNull(this.list.getNext());
+		
+		Order orderThree = createOrder("three");
+		this.list.addItem(orderThree);
+
+		// for now the pointer is on null.
```

Comment:
I don't understand this comment. What you mean by "pointer"?  Do you mean this.current? addItem should fix this.current to point to this.tail, when this.current == null (i.e. this.current is beyond the last node of the list). So, the comment as it is would make since if it appeared before this.list.addItem(orderThree). Here, the correct comment would be: // addItem() should have fixed this.current to point to the newly item added to the tail of the list.

Reply:
Sure, it was a doubt we have discussed yesterday, and I forgot to change this comment to match the changes we've made to addItem().